src/mumble/Log.cpp

```diff
@@ -722,7 +722,10 @@ void Log::log(MsgType mt, const QString &console, const QString &terse, bool own
 
 		LogTextBrowser *tlog     = Global::get().mw->qteLog;
 		const int oldscrollvalue = tlog->getLogScroll();
-		const bool scroll        = (oldscrollvalue == tlog->getLogScrollMaximum());
+		// Restore the previous scroll position after inserting a new message
+		// if the message was not sent by the user AND the chat log is not
+		// scrolled all the way down.
+		const bool restoreScroll = !ownMessage && !tlog->isScrolledToBottom();
```

Comment:
```suggestion
		const bool restoreScroll = !(ownMessage || tlog->isScrolledToBottom());
```
I think this form is easier parsable by fellow humans :)